File: Python/my_service.py
from single_sign_on import SingleSignOnRegistry, SSOToken


class MyService:
    
    def __init__(self, sso_registry):
        self.sso_registry = sso_registry
    
    def handle_request(self, request):
        if not self.sso_registry.is_valid(request.token):
            return Response("unauthorized")
        return Response("Hello {0}!".format(request.name))


class Request:
    def __init__(self, name, token):
        self.name = name
        self.token = token


class Response:
    def __init__(self, text):
        self.text = text

File: TypeScript/my_service.ts
import { SingleSignOnRegistry, SSOToken} from "./single_sign_on";

export class Request {
    name: string
    token: SSOToken
    constructor(name: string, token: SSOToken) {
        this.name = name
        this.token = token
    }
}

export class Response{
    text:string
    constructor(text:string) {
        this.text = text
    }
}

export class MyService {

    sso_registry: SingleSignOnRegistry


    constructor(sso_registry: SingleSignOnRegistry) {
        this.sso_registry = sso_registry
    }

    handle_request(request: Request): Response {
        // TODO: check token against SSORegistry
        return new Response(`Hello ${request.name}!`)
    }
}
